### PR TITLE
[HBase] Add HBase Master/RegionServer Integration

### PR DESCRIPTION
**Pull request by everpeace**
Hi, I have operated HBase clusters for this several months and monitored the clusters with DataDog.
I'm happy to contribute our jmx configuration for HBase Master/Regionserver to the dd-agent extras integration package.

HBase Master and RegionServer exposes bunch of MBeans for monitoring.  So, we filtered some important metrics in terms of performance and cluster health.   For details, see [hbase_master/conf.yaml.example](https://github.com/everpeace/integrations-extras/blob/hbase/hbase_master/conf.yaml.example#L23) and [hbase_regionserver/conf.yaml.example](https://github.com/everpeace/integrations-extras/tree/hbase/hbase_regionserver/conf.yaml.example#L23).

Implementation is really simple.  `conf.yaml.example` is similar to usual `jmx.yaml`.  The differences are 
- `init_config.is_jmx: true`
- All MBean filters are in `init_config.conf` rather than `inistances[].conf`

__Note:__ most of committed code is testing codes.  integration tests spawns hbase cluster in pseudo-distributed mode by docker-compose and verifies datadog-agent can collect jmx metrics actually.

I'm wondering one thing.  I have [value conversion setting in HBase Master metrics](https://github.com/everpeace/integrations-extras/blob/hbase/hbase_master/conf.yaml.example#L93-L99).  I wrote this config by refering [test of jmxfetch](https://github.com/DataDog/jmxfetch/blob/0.13.1/src/test/resources/jmx.yaml#L33-L38).  However this doesn't work.  So, I ignored this metrics in [test](https://github.com/everpeace/integrations-extras/blob/hbase/hbase_master/test_hbase_master.py#L101-L103)
 
I'm really appreciated for comments or feedbacks :-)

----

Screenshots of sample timeboards are here:
- HBase Master
   image
- HBase Region Server
   image

### CODE REVIEW

**Comment by everpeace** [comment]
@hush-hush sorry for bothering though, would you mind me asking you roughly when you would take a look at this PR please??

**Comment by hush-hush** [comment]
Hi @everpeace, I'm sorry for the delay. We are currently finishing the ground rules and documentation on how to handle integration in extras. We hope to update the documentation in the README in the next weeks.

I took a quick first pass and it looks great.Thanks for the contribution !

The idea behind integration-extras is to have integrations maintain by the community.
The main idea it to have a maintainer that actually uses the integrations and knows the underlying software (HBase in our case).

Would you mind being the maintainer for HBaseMaster and HBaseRegionServer ?

You can step down at any moment of course, the integration is then marked as ‘orphaned’ and someone can step up to take over. The main task of a maintainer from integrations-extras would be to handle PR and issues for his/her integration when they have time.

If this is not something you would like to, we could add a link (to your repository/branch) to the list of unofficial integrations (integration available but not maintain by anybody).

If you have any questions, I or @irabinovitch  would be glad to answer them ! This is the very beginning of integrations-extras and we are still setting up everything on our side.

**Comment by everpeace** [comment]
@hush-hush cc/ @irabinovitch Thank you for your response and sharing updates!  I understand this repo's status.

> Would you mind being the maintainer for HBaseMaster and HBaseRegionServer?

Not at all.  I'm really happy to be a maintainer of HBase Integration module.

**Comment by gmmeyer** [comment]
We'll do a more thorough look through later, but we're actually using Travis instead of Circle. The circle.yml should be removed. Can you add those to the Travis matrix instead of the circle one? I can tell you what to do if you're not sure! 😄

**Comment by everpeace** [comment]
@gmmeyer Thanks for pointing out!  I added `ci:run[hbase_master]` and `ci:run[hbase_regionserver]` to `.travis.yml`.

File: hbase_master/conf.yaml.example [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    # name: hbase_master
+    # # java_bin_path: /path/to/java # Optional, should be set if the agent cannot find your java executable
+    # # java_options: "-Xmx200m -Xms50m" # Optional, Java JVM options
+    # # trust_store_path: /path/to/trustStore.jks # Optional, should be set if ssl is enabled
+    # # trust_store_password: password
+
+init_config:
+  is_jmx: true
+
+  # Metrics collected by this check. You should not have to modify this.
```
Comment (hush-hush) [inline hbase_master/conf.yaml.example]: Could you add comment for each attribute so people know what they are collecting and why it's interesting ?

File: hbase_master/manifest.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "maintainer": "help@datadoghq.com",
```
Comment (hush-hush) [inline hbase_master/manifest.json]: Could you update the maintainer field with your github user or mail ?

File: hbase_master/metadata.csv [hunk 3]
```diff
@@ -0,0 +1 @@
+metric_name,metric_type,interval,unit_name,per_unit_name,description,orientation,integration,short_name
```
Comment (hush-hush) [inline hbase_master/metadata.csv]: Please update the csv to list the metrics collected (see more here: http://docs.datadoghq.com/guides/integration_sdk/#metadatacsv)

File: hbase_regionserver/conf.yaml.example [hunk 4]
```diff
@@ -0,0 +1,530 @@
...
+    # name: hbase_master
+    # # java_bin_path: /path/to/java # Optional, should be set if the agent cannot find your java executable
+    # # java_options: "-Xmx200m -Xms50m" # Optional, Java JVM options
+    # # trust_store_path: /path/to/trustStore.jks # Optional, should be set if ssl is enabled
+    # # trust_store_password: password
+
+init_config:
+  is_jmx: true
+
+  # Metrics collected by this check. You should not have to modify this.
```
Comment (hush-hush) [inline hbase_regionserver/conf.yaml.example]: Could you add comment for each attribute so people know what they are collecting and why it's interesting ?
Or at least the main ones.

File: hbase_regionserver/manifest.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
+  "maintainer": "help@datadoghq.com",
```
Comment (hush-hush) [inline hbase_regionserver/manifest.json]: Same here

File: hbase_regionserver/metadata.csv [hunk 6]
```diff
@@ -0,0 +1 @@
+metric_name,metric_type,interval,unit_name,per_unit_name,description,orientation,integration,short_name
```
Comment (hush-hush) [inline hbase_regionserver/metadata.csv]: Could you also list the hbase_regionserver metrics here ?

**Comment by gmmeyer** [comment]
@everpeace I hate to do this, because it was our fault, but after you fixed it, I noticed that the tests weren't running. The fix was incorrect. You'll need to rebase to master, because the mistake was borne from one of our mistakes in turn, and then add this to the matrix above:

```
    - TRAVIS_FLAVOR=hbase_master
    - TRAVIS_FLAVOR=hbase_regionserver
```

**Comment by everpeace** [comment]
@gmmeyer Thanks for your guidance!  I rebased my branch and update `.travis.yml`  with travis matrix!

**Comment by everpeace** [comment]
@hush-hush Thanks for your comments!  

updating `manifest.json` is done.
I'm doing updating `metadata.csv` and `config.yaml.example` for having more description.  
( In this process, I introduced `_gen.rb` for generating `metatdata.csv` and `config.yaml.example` with description which are extracted actual hbase jmx mbeans.)

After finishing up, I will contact you.

**Comment by everpeace** [comment]
@hush-hush I updated `metadata.csv` and comments in `conf.yaml.example` with the help of my friend @iwasakims who is one of Hadoop PMC members.

Several metrics lacks descriptions but those names are  intuitive enough and typical users who wants to install this integration module would not need those descriptions.

I'm glad you to take a look at this PR.  Thanks in advance!

**Comment by hush-hush** [comment]
@everpeace Thanks for your patience and contribution. I just merged the PR !

**Comment by everpeace** [comment]
@hush-hush Wow, thank you so much!! I'm glad to contribute datadog integration community.

**Pull request merged**